### ISSUE TITLE
I2S_FIFO_WR_REG & I2S_FIFO_WR_REG undefined (IDFGH-7273)

### ISSUE THREAD

**Issue by HoverClub** [espressif/esp-idf#8862]
### Description

As it says in the title the I2S_FIFO_WR_REG & I2S_FIFO_WR_REG aren't defined in '''i2s_struct.h'''.  These regs are documented in the tech ref manual 12.7 and actually work correctly if added to i2s_struct.h:
```cpp
typedef volatile struct i2s_dev_s {
    //uint32_t reserved_0;
    //uint32_t reserved_4;
    uint32_t fifo_wr;
    uint32_t fifo_rd;
    union {
        struct {
```

### Debug Message

```plain
Compiling .pio/build/wemos_d1_mini32/src/testfifo1.cpp.o
src/testfifo1.cpp: In function 'void i2s_init()':
src/testfifo1.cpp:189:10: error: 'i2s_dev_t' {aka 'volatile struct i2s_dev_s'} has no member named 'fifo_wr'
     I2S0.fifo_wr = 0x44993388; // init some data for testing
          ^~~~~~~~~~~
*** [.pio/build/wemos_d1_mini32/src/testfifo1.cpp.o] Error 1
```

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by Alvin1Zhang** [comment]
Thanks for reporting. Would you please help provide more details as suggested in the [issue template](https://github.com/espressif/esp-idf/issues/new?assignees=&labels=&template=bug_report.md&title=)? Information like elf, sdk configuration, backtrace, log outputs, commit ID, hardware and etc. would help us debug further. Thanks.

**Comment by HoverClub** [comment]
I'm using the Arduino framework with the file i2s_struct.h which is the missing defines for the I2S_FIFO_WR_REG and I2S_FIFO_RD_REG detailed in the Tech manual 12.7, etc.  Hardware is irrelevant as its' a framework issue.  IDF file is here: https://github.com/espressif/esp-idf/blob/master/components/soc/esp32/include/soc/i2s_struct.h
it has these reg locations defined as reserved.

**Comment by L-KAYA** [comment]
Thank for reporting this issue, I checked both `i2s_struct.h` and `i2s_reg.h` and also did a test on that two fields, I think this two  fields are probably dropped while generating header files, it will be added back.

**Comment by Alvin1Zhang** [comment]
Thanks for reporting, fix on master branch is available at https://github.com/espressif/esp-idf/commit/94d7ae03da0d00fcfc81c2e384e36f82b84584ef, and we will also backport the fix to release/4.4, thanks.
